### DISCUSSION TITLE
学んだこと、つまったことを自分で解決した場合の共有場所

### DISCUSSION THREAD

**Discussion by pss-aileen** [first-contributions-ja/first-contributions-ja.github.io#25] [【運営】相談・報告]
@kazzyfrog @kagomen kanaさん
ご意見ください〜！

---

- https://github.com/first-contributions-ja/first-contributions-ja/issues/9#issuecomment-1989975201
- こちらにも書きましたが、誰か一人が悩んだことは誰かの助けになるというか、自分が悩んだことは誰かも悩んだことがある可能性がある
- どこかに悩んだ過程や試したこと、最終的な結論などをまとめられれば良いなと思った
- どうまとめるか
  - **Discussionsのカテゴリーに「知見・知識〜〜」というものを作ったのでそこにまとめる**
    - と思ったが、なんか後々探すのが大変そう
  - **専用のリポジトリを作る**
    - issueにて各々が書ける簡易的なブログ？レポートのようにする。ある程度フォーマットを決める
    - 例）目的: git push origin HEAD と head の違いを検証、どちらを使うべきか考える。過程、調査: 〜〜〜、結論: HEADを使うべき→みんなどう思う？そうだね！→コントリビューションの書き換えが必要→Issueをたてる
    - ↑こういった感じ
    - GitHubのIssueは画像ぺこぺこはれるし、コメントをどんどん残していけるので経過をたどりやすく個人的に最適かなと考えている
    - 自分たち以外で学びたい人、チーム運営したい人、技術面でわからないことがある人の助けになるように意識
    - わからないこと、技術面で悩んだことを質問したい時もこっちのIssueにすればいいかも

**Comment by kagomen** [comment]
そういう場所があればと思っていたので、賛成です！
自分はDiscussionsで良いんじゃないかなーという気がするのですが、issueと比べて
> なんか後々探すのが大変そう

なんですかね❔

**Reply by pss-aileen** [reply]
これ書いた後にDiscussionsでもやっぱり良いかな〜と迷い始めたんですが...

探すのが大変そうっていうのは
Discussionsに入れると技術の疑問と、みんなとただ議論したい内容が混在するとなんだかなぁ〜と
今後たくさん共有することが増えて、Discussionsが増えるとごちゃごちゃしそうだな...と感じまして。

その他リポジトリを分ける利点として
- GitHub・Next.js、JavaScript、React、Gitなど、ここのリポと分けたとラベルをつけることができる
  - 開発のリポのラベルと混在させるとissueのラベルつけるときに沢山あって悩みそう
- 技術で悩んだときはこのリポジトリを開いて書けばいい、ログとして残せばいいとシンプルになりそう

といった感じですね！
とりあえずDiscussionで運用してみて、やっぱり...と思ったらリポジトリ作ってもよいかなとも思います。

あーでもDiscussionの利点としても
コメントすると上部に上がってくるっていうのがあるので、最新の悩み事が共有されやすくて良いなとも感じます！

最終的には好みとかにもなりそうですが、現状私はどっちでも良いかなという気分になりつつあります...！！

**Reply by pss-aileen** [reply]
> あーでもDiscussionの利点としても
> コメントすると上部に上がってくるっていうのがあるので、最新の悩み事が共有されやすくて良いなとも感じます！

これ、一応issueでもソートすれば見れました！

**Reply by kagomen** [reply]
理解しました。
Disucussionsと別リポジトリのIssuesの差異としては、
Discussionsでカテゴリとラベルで検索する手間と、別リポジトリのIssuesに移動する手間、って感じですかね❔

個人的にはある程度雑多になっても、
- 行動するまでのハードルが低い方がいい
- 投稿が集まる→盛り上がってるように見える

という点でDiscussions推し（？）ですが、こういうのは好みですよね。

**Reply by pss-aileen** [reply]
ですね...！
とりあえずDiscussionsでやってみますか！

**Comment by kazzyfrog** [comment]
> とりあえずDiscussionsでやってみますか！

了解です！👍

ジャストアイデアですが、知見をマークダウンファイルで管理できたら、サイト上（first-contributions-ja.github.io）で公開したりも出来そうだなと思いました！
他にも、このリポジトリのWikiとかを使ってもいいかも！

**Reply by pss-aileen** [reply]
知見のまとめ方を考えていたのですが
Qiitaに組織申請して作って、各々のアカウントでノウハウを書くことができれば
個人で学んだこととしても投稿ができるし、このリポジトリの宣伝もできそうで良いなぁ〜と思ったりしました。

[Qiita Organizationについて - Qiita](https://qiita.com/organizations/about)

記事書くとかなり労力必要なので後回しでいいかなとは思いますが、ふと思いついたので書きました

---

別件ですが、色々調べたら [Qiita Team - かんたんにナレッジを共有できる社内向け情報共有サービス](https://teams.qiita.com/) こういうのがありました。
さすがに会社ではないので導入はしなくて良いかなとはありますが、こういったものもあるんだ〜と勉強になりました。

**Reply by kagomen** [reply]
良さそうですね！やってみたいです👀
ちなみにZennでも似たような機能があるみたいです。
[Publicationの使い方 -Zenn](https://zenn.dev/zenn/articles/how-to-use-publication)

**Reply by kagomen** [reply]
Zennの方で試しに作ってみようと思ったのですが、以下の結果になりました💧
<img width="521" alt="スクリーンショット 2024-03-20 0 42 18" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/154225199/0f0de8f5-bd8d-4d65-bcc2-77066ae13dc8">

そこでふと、こういうのってプロジェクト名ではなくチーム名で登録するんだろうなーと思いました。
以前にも話題に上がってたと思うのですが、やはりチーム名があると便利ですよね……！

**Comment by pss-aileen** [comment]
使い心地メモ

### Discussions
- 良いところ
- 気になるところ
  - Labelをissueのように一括でつけられない
